docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+When importing an identity, a Device Key Pair will be created and its public key added to the DID-Document as well. Depending on the DID-method, the Master Private Key might be needed to update the DID-Document.
+
+Some DID-methods might support granular permissions. This enables users to grant a specific set of permissions to different devices based on how secure the device is and how likely it is to be lost or robbed.
+
+#### Managing verifiable credentials
+
+Users will be able to view, add and remove Verifiable Credentials to and from their wallet. As we previously explained in the [Verifiable Credentials](#verifiable-credentials) section, the IDM Wallet will initially focus on self-issued credentials and expand to credentials issued by others when the ecosystem gets mature.
+
+Depending on the identities' DID-methods and their capabilities, where and how the credentials are stored might be pre-defined by the DID-method and may be either off-chain or on-chain. As an example, [EIP780](https://github.com/ethereum/EIPs/issues/780) proposes an on-chain registry for Ethereum based identities to store credentials, enabling persons, smart contracts, and machines to issue credentials about each other. The IDM Wallet will employ best-practices for each DID-method when storing credentials.
+
+#### Authentication on applications
+
+Most applications need to know the identity of the user so that they can provide a customized experience.
+
+Applications may start an authentication process, issuing an authentication request to an IDM Wallet, containing the application DID and Verifiable Credentials with its details, such as the application name, and the credentials it wants to receive from an identity. The IDM Wallet will perform DID-Auth to ensure that the application is in control of the DID it presented. If successful, the user is then prompted to select an identity and to accept what the application is asking for.
+
+If the user accepted the prompt, a new session is created and will be used for any further interactions between that application and that IDM Wallet. The session is composed of a key-pair where the public key is known by both, serving as the session identifier, while the private key is only known by that IDM Wallet. Additionally, all sessions have a max-age, meaning they expire as time passes.

```

Comment:
Wouldn’t it make sense to state here that the public session key is signed by the device key where the session was created?

Reply:
What do you think of:

> If the user accepted the prompt, a new session is created and will be used for any further interactions between that application and that IDM Wallet. The session is composed of a key-pair where the public key is known by both, serving as the session identifier, while the private key is only known by that IDM Wallet. Moreover, the Session Public Key was signed with the Device Private Key, serving as an authorization proof. Worth noting that all sessions have a max-age, meaning they expire as time passes.

Reply:
It seems way better to me, now! 😝